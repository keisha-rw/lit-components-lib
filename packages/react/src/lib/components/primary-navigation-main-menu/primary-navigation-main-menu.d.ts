import * as React from 'react';
export declare const PdsPrimaryNavigationMainMenu: React.FunctionComponent<Partial<{
    ariaLabel: string;
    variant: "inverted" | "default";
}> & {
    ref?: React.Ref<any> | undefined;
} & {
    style?: React.CSSProperties | undefined;
} & Partial<{
    onMainMenuDropdownOpen: (e: Event) => void;
    onMainMenuDropdownClose: (e: Event) => void;
    onMainMenuLinkClick: (e: Event) => void;
}> & {
    id?: string | undefined;
    slot?: string | undefined;
    className?: string | undefined;
    children?: React.ReactNode;
}>;
