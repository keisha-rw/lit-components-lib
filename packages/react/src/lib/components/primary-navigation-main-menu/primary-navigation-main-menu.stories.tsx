import * as React from 'react';
import type { StoryFn } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import { PdsPrimaryNavigationMainMenu } from './primary-navigation-main-menu';
import { PdsPrimaryNavigationMainMenuItem } from '../primary-navigation-main-menu-item/primary-navigation-main-menu-item';
import { PdsList } from '../list/list';
import { PdsListItem } from '../list-item/list-item';
import { PdsGrid } from '../grid/grid';
import { PdsPrimaryNavigationDropdownLink } from '../primary-navigation-dropdown-link/primary-navigation-dropdown-link';
import { PdsGridItem } from '../grid-item/grid-item';
import { PlaceholderElement } from '../../../../.storybook/components/placeholder-element/placeholder-element';

export default {
  title: 'Components/Primary navigation/Primary navigation main menu',
  component: PdsPrimaryNavigationMainMenu,
  tags: ['autodocs'],
  parameters: {
    happo: false,
    docs: {
      description: {
        component:
          'Note: This component is ONLY meant to be used as a subcomponent of the primary navigation component.',
      },
    },
  },
};

const Template: StoryFn<typeof PdsPrimaryNavigationMainMenu> = () => (
  <PdsPrimaryNavigationMainMenu
    slot="primary-navigation-main-menu"
    aria-label="Main navigation area"
  >
    <PdsPrimaryNavigationMainMenuItem
      dropdown
      text="Dropdown"
      onDropdownOpen={(e: any) =>
        action('pds-primary-navigation-main-menu-dropdown-open')(e.detail)
      }
      onDropdownClose={(e: any) =>
        action('pds-primary-navigation-main-menu-dropdown-close')(e.detail)
      }
    >
      <PdsList size="sm" spacing="sm">
        <PdsListItem>
          <PdsPrimaryNavigationDropdownLink href="#">
            This is a link
          </PdsPrimaryNavigationDropdownLink>
        </PdsListItem>
        <PdsListItem>
          <PdsPrimaryNavigationDropdownLink href="#">
            This is a link
          </PdsPrimaryNavigationDropdownLink>
        </PdsListItem>
        <PdsListItem>
          <PdsPrimaryNavigationDropdownLink href="#">
            This is a link
          </PdsPrimaryNavigationDropdownLink>
        </PdsListItem>
        <PdsListItem>
          <PdsPrimaryNavigationDropdownLink href="#">
            This is a link
          </PdsPrimaryNavigationDropdownLink>
        </PdsListItem>
      </PdsList>
    </PdsPrimaryNavigationMainMenuItem>
    <PdsPrimaryNavigationMainMenuItem
      megamenu
      text="Megamenu"
      onDropdownOpen={(e: any) =>
        action('pds-primary-navigation-main-menu-dropdown-open')(e.detail)
      }
      onDropdownClose={(e: any) =>
        action('pds-primary-navigation-main-menu-dropdown-close')(e.detail)
      }
    >
      <PdsGrid variant="1-3up" break="slower">
        <PdsGridItem>
          <PlaceholderElement styleModifier="pds-u-margin-bottom-0">
            Item 1
          </PlaceholderElement>
        </PdsGridItem>
        <PdsGridItem>
          <PlaceholderElement styleModifier="pds-u-margin-bottom-0">
            Item 2
          </PlaceholderElement>
        </PdsGridItem>
        <PdsGridItem>
          <PlaceholderElement styleModifier="pds-u-margin-bottom-0">
            Item 3
          </PlaceholderElement>
        </PdsGridItem>
        <PdsGridItem>
          <PlaceholderElement styleModifier="pds-u-margin-bottom-0">
            Item 4
          </PlaceholderElement>
        </PdsGridItem>
        <PdsGridItem>
          <PlaceholderElement styleModifier="pds-u-margin-bottom-0">
            Item 5
          </PlaceholderElement>
        </PdsGridItem>
        <PdsGridItem>
          <PlaceholderElement styleModifier="pds-u-margin-bottom-0">
            Item 6
          </PlaceholderElement>
        </PdsGridItem>
      </PdsGrid>
    </PdsPrimaryNavigationMainMenuItem>
    <PdsPrimaryNavigationMainMenuItem
      href="#"
      text="Link"
      onClick={(e: any) =>
        action('pds-primary-navigation-main-menu-link-click')(e.detail)
      }
    />
  </PdsPrimaryNavigationMainMenu>
);
export const Default = Template.bind({});
Default.args = {};

const LinkWithIconTemplate: StoryFn<
  typeof PdsPrimaryNavigationMainMenu
> = () => (
  <PdsPrimaryNavigationMainMenu
    slot="primary-navigation-main-menu"
    aria-label="Main navigation area"
  >
    <PdsPrimaryNavigationMainMenuItem
      dropdown
      text="Dropdown"
      onDropdownOpen={(e: any) =>
        action('pds-primary-navigation-main-menu-dropdown-open')(e.detail)
      }
      onDropdownClose={(e: any) =>
        action('pds-primary-navigation-main-menu-dropdown-close')(e.detail)
      }
    >
      <PdsList size="sm" spacing="sm">
        <PdsListItem>
          <PdsPrimaryNavigationDropdownLink href="#">
            This is a link
          </PdsPrimaryNavigationDropdownLink>
        </PdsListItem>
        <PdsListItem>
          <PdsPrimaryNavigationDropdownLink href="#">
            This is a link
          </PdsPrimaryNavigationDropdownLink>
        </PdsListItem>
        <PdsListItem>
          <PdsPrimaryNavigationDropdownLink href="#">
            This is a link
          </PdsPrimaryNavigationDropdownLink>
        </PdsListItem>
        <PdsListItem>
          <PdsPrimaryNavigationDropdownLink href="#">
            This is a link
          </PdsPrimaryNavigationDropdownLink>
        </PdsListItem>
      </PdsList>
    </PdsPrimaryNavigationMainMenuItem>
    <PdsPrimaryNavigationMainMenuItem
      megamenu
      text="Megamenu"
      onDropdownOpen={(e: any) =>
        action('pds-primary-navigation-main-menu-dropdown-open')(e.detail)
      }
      onDropdownClose={(e: any) =>
        action('pds-primary-navigation-main-menu-dropdown-close')(e.detail)
      }
    >
      <PdsGrid variant="1-3up" break="slower">
        <PdsGridItem>
          <PlaceholderElement styleModifier="pds-u-margin-bottom-0">
            Item 1
          </PlaceholderElement>
        </PdsGridItem>
        <PdsGridItem>
          <PlaceholderElement styleModifier="pds-u-margin-bottom-0">
            Item 2
          </PlaceholderElement>
        </PdsGridItem>
        <PdsGridItem>
          <PlaceholderElement styleModifier="pds-u-margin-bottom-0">
            Item 3
          </PlaceholderElement>
        </PdsGridItem>
        <PdsGridItem>
          <PlaceholderElement styleModifier="pds-u-margin-bottom-0">
            Item 4
          </PlaceholderElement>
        </PdsGridItem>
        <PdsGridItem>
          <PlaceholderElement styleModifier="pds-u-margin-bottom-0">
            Item 5
          </PlaceholderElement>
        </PdsGridItem>
        <PdsGridItem>
          <PlaceholderElement styleModifier="pds-u-margin-bottom-0">
            Item 6
          </PlaceholderElement>
        </PdsGridItem>
      </PdsGrid>
    </PdsPrimaryNavigationMainMenuItem>
    <PdsPrimaryNavigationMainMenuItem
      href="#"
      text="Link"
      arrow
      onClick={(e: any) =>
        action('pds-primary-navigation-main-menu-link-click')(e.detail)
      }
    />
  </PdsPrimaryNavigationMainMenu>
);
export const LinkWithIcon = LinkWithIconTemplate.bind({});
LinkWithIcon.args = {};
LinkWithIcon.storyName = 'Link with icon';
