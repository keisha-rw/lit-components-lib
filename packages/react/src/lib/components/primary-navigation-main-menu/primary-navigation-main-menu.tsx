import * as React from 'react';
import { PdsPrimaryNavigationMainMenu as PdsPrimaryNavigationMainMenuElement } from '@keisha/design-system';
import { EventName } from '@lit-labs/react';
import { PdsPrimaryNavigationMainMenuProps } from '../../built-component-props';
import { createPdsReactComponent } from '../../create-pds-react-component';

export const PdsPrimaryNavigationMainMenu =
  createPdsReactComponent<PdsPrimaryNavigationMainMenuProps>()({
    tagName: 'pds-primary-navigation-main-menu',
    elementClass: PdsPrimaryNavigationMainMenuElement,
    react: React,
    events: {
      onMainMenuDropdownOpen:
        'pds-primary-navigation-main-menu-dropdown-open' as EventName<CustomEvent>,
      onMainMenuDropdownClose:
        'pds-primary-navigation-main-menu-dropdown-close' as EventName<CustomEvent>,
      onMainMenuLinkClick:
        'pds-primary-navigation-main-menu-link-click' as EventName<CustomEvent>,
    },
  });
