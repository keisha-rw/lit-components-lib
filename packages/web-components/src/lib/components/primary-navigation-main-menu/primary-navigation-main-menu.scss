/*------------------------------------*\
 #MAIN_NAV
\*------------------------------------*/

.pds-c-primary-navigation-main-menu {
  float: none;
  display: block;
  padding: size(2) size(3) 0 size(3);

  @media all and (min-width: $pds-bp-lg) {
    float: left;
    display: inline;
    padding: size(1.5);
  }

  @media all and (width < $pds-bp-lg) {
    padding: size(3) size(3) 0 size(3);
  }
}

// Default mobile nav
.pds-c-primary-navigation-main-menu--default {
  // Submenu styles
  --pds-primary-navigation-main-menu-item-link-background: var(
    --pds-component-primary-nav-default-mobile-submenu-background
  );
  --pds-primary-navigation-main-menu-item-link-color: var(
    --pds-component-primary-nav-default-mobile-submenu-label
  );
  --pds-primary-navigation-main-menu-item-icon-color: var(
    --pds-component-primary-nav-default-mobile-submenu-icon
  );
  --pds-primary-navigation-main-menu-item-link-hover-background: var(
    --pds-component-primary-nav-default-mobile-submenu-hover-background
  );
  --pds-primary-navigation-main-menu-item-link-hover-color: var(
    --pds-component-primary-nav-default-mobile-submenu-hover-label
  );
  --pds-primary-navigation-main-menu-item-icon-hover-color: var(
    --pds-component-primary-nav-default-mobile-submenu-hover-icon
  );
  --pds-primary-navigation-main-menu-item-link-active-background: var(
    --pds-component-primary-nav-default-mobile-submenu-active-background
  );
  --pds-primary-navigation-main-menu-item-link-active-color: var(
    --pds-component-primary-nav-default-mobile-submenu-active-label
  );
  --pds-primary-navigation-main-menu-item-icon-active-color: var(
    --pds-component-primary-nav-default-mobile-submenu-active-icon
  );

  // Links below submenu
  --pds-primary-navigation-main-menu-item-href-color: var(
    --pds-component-primary-nav-default-mobile-link
  );
  --pds-primary-navigation-main-menu-item-href-hover-color: var(
    --pds-component-primary-nav-default-mobile-hover-link
  );

  // Default desktop styles
  @media all and (min-width: $pds-bp-lg) {
    --pds-primary-navigation-main-menu-item-dropdown-background: var(
      --pds-component-primary-nav-default-dropdown-background
    );
    --pds-primary-navigation-main-menu-item-dropdown-arrow-top-position: 160%;
    --pds-primary-navigation-main-menu-item-link-background: var(
      --pds-component-primary-nav-default-nav-item-background
    );
    --pds-primary-navigation-main-menu-item-link-color: var(
      --pds-component-primary-nav-default-nav-item-text
    );
    --pds-primary-navigation-main-menu-item-icon-color: var(
      --pds-component-primary-nav-default-nav-item-icon
    );
    --pds-primary-navigation-main-menu-item-divider: var(
      --pds-component-primary-nav-default-nav-item-divider
    );
    --pds-primary-navigation-main-menu-item-link-hover-background: var(
      --pds-component-primary-nav-default-nav-item-hover-background
    );
    --pds-primary-navigation-main-menu-item-link-hover-color: var(
      --pds-component-primary-nav-default-nav-item-hover-text
    );
    --pds-primary-navigation-main-menu-item-icon-hover-color: var(
      --pds-component-primary-nav-default-nav-item-hover-icon
    );
    --pds-primary-navigation-main-menu-item-link-active-background: var(
      --pds-component-primary-nav-default-nav-item-active-background
    );
    --pds-primary-navigation-main-menu-item-link-active-color: var(
      --pds-component-primary-nav-default-nav-item-active-text
    );
    --pds-primary-navigation-main-menu-item-icon-active-color: var(
      --pds-component-primary-nav-default-nav-item-active-icon
    );
  }
}

// Inverted mobile styles
.pds-c-primary-navigation-main-menu--inverted {
  // Submenu styles
  --pds-primary-navigation-main-menu-item-link-background: var(
    --pds-component-primary-nav-inverted-mobile-submenu-background
  );
  --pds-primary-navigation-main-menu-item-link-color: var(
    --pds-component-primary-nav-inverted-mobile-submenu-label
  );
  --pds-primary-navigation-main-menu-item-icon-color: var(
    --pds-component-primary-nav-inverted-mobile-submenu-icon
  );
  --pds-primary-navigation-main-menu-item-link-hover-background: var(
    --pds-component-primary-nav-inverted-mobile-submenu-hover-background
  );
  --pds-primary-navigation-main-menu-item-link-hover-color: var(
    --pds-component-primary-nav-inverted-mobile-submenu-hover-label
  );
  --pds-primary-navigation-main-menu-item-icon-hover-color: var(
    --pds-component-primary-nav-inverted-mobile-submenu-hover-icon
  );
  --pds-primary-navigation-main-menu-item-link-active-background: var(
    --pds-component-primary-nav-inverted-mobile-submenu-active-background
  );
  --pds-primary-navigation-main-menu-item-link-active-color: var(
    --pds-component-primary-nav-inverted-mobile-submenu-active-label
  );
  --pds-primary-navigation-main-menu-item-icon-active-color: var(
    --pds-component-primary-nav-inverted-mobile-submenu-active-icon
  );

  // Links below submenu
  --pds-primary-navigation-main-menu-item-href-color: var(
    --pds-component-primary-nav-inverted-mobile-link
  );
  --pds-primary-navigation-main-menu-item-href-hover-color: var(
    --pds-component-primary-nav-inverted-mobile-hover-link
  );

  // Inverted desktop styles
  @media all and (min-width: $pds-bp-lg) {
    --pds-primary-navigation-main-menu-item-dropdown-background: var(
      --pds-component-primary-nav-inverted-dropdown-background
    );
    --pds-primary-navigation-main-menu-item-dropdown-arrow-top-position: 160%;
    --pds-primary-navigation-main-menu-item-link-background: var(
      --pds-component-primary-nav-inverted-nav-item-background
    );
    --pds-primary-navigation-main-menu-item-link-color: var(
      --pds-component-primary-nav-inverted-nav-item-text
    );
    --pds-primary-navigation-main-menu-item-icon-color: var(
      --pds-component-primary-nav-inverted-nav-item-icon
    );
    --pds-primary-navigation-main-menu-item-divider: var(
      --pds-component-primary-nav-inverted-nav-item-divider
    );
    --pds-primary-navigation-main-menu-item-link-hover-background: var(
      --pds-component-primary-nav-inverted-nav-item-hover-background
    );
    --pds-primary-navigation-main-menu-item-icon-hover-color: var(
      --pds-component-primary-nav-inverted-nav-item-hover-icon
    );
    --pds-primary-navigation-main-menu-item-link-hover-color: var(
      --pds-component-primary-nav-inverted-nav-item-hover-text
    );
    --pds-primary-navigation-main-menu-item-link-active-background: var(
      --pds-component-primary-nav-inverted-nav-item-active-background
    );
    --pds-primary-navigation-main-menu-item-link-active-color: var(
      --pds-component-primary-nav-inverted-nav-item-active-text
    );
    --pds-primary-navigation-main-menu-item-icon-active-color: var(
      --pds-component-primary-nav-inverted-nav-item-active-icon
    );
  }
}

.pds-c-primary-navigation-main-menu__list {
  display: flex;
  flex-direction: column;
  gap: size(1);
  margin: 0;
  padding: 0;
  list-style: none;

  @media all and (min-width: $pds-bp-lg) {
    flex-direction: row;
  }
}
