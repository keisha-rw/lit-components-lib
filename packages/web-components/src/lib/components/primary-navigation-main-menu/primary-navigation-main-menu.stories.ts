import { html } from 'lit';
import './primary-navigation-main-menu';
import { Meta, StoryObj } from '@storybook/web-components';
import '../primary-navigation-main-menu-item/primary-navigation-main-menu-item';
import '../list/list';
import '../list-item/list-item';
import '../grid/grid';
import '../grid-item/grid-item';
import '../primary-navigation-dropdown-link/primary-navigation-dropdown-link';
import '../../../../.storybook/components/placeholder-element/placeholder-element';

const meta: Meta = {
  title: 'Components/Primary navigation/Primary navigation main menu',
  component: 'pds-primary-navigation-main-menu',
  tags: ['autodocs'],
  parameters: {
    happo: false,
    actions: {
      handles: [
        'pds-primary-navigation-item-click',
        'pds-primary-navigation-main-menu-dropdown-close',
        'pds-primary-navigation-main-menu-dropdown-open',
        'pds-primary-navigation-main-menu-link-click',
      ],
    },
    docs: {
      description: {
        component:
          'Note: This component is ONLY meant to be used as a subcomponent slotted within the primary navigation component.',
      },
    },
  },
  render: (args) =>
    html` <pds-primary-navigation-main-menu
      variant="${args.variant}"
      ariaLabel=${args.ariaLabel}
    >
      <pds-primary-navigation-main-menu-item dropdown text="Dropdown">
        <pds-list size="sm" spacing="sm">
          <pds-list-item
            ><pds-primary-navigation-dropdown-link href="#"
              >This is a link</pds-primary-navigation-dropdown-link
            ></pds-list-item
          >
          <pds-list-item
            ><pds-primary-navigation-dropdown-link href="#"
              >This is a link</pds-primary-navigation-dropdown-link
            ></pds-list-item
          >
          <pds-list-item
            ><pds-primary-navigation-dropdown-link href="#"
              >This is a link</pds-primary-navigation-dropdown-link
            ></pds-list-item
          >
          <pds-list-item
            ><pds-primary-navigation-dropdown-link href="#"
              >This is a link</pds-primary-navigation-dropdown-link
            ></pds-list-item
          >
        </pds-list>
      </pds-primary-navigation-main-menu-item>
      <pds-primary-navigation-main-menu-item megamenu text="Megamenu">
        <pds-grid variant="1-3up" break="slower">
          <pds-grid-item>
            <placeholder-element>Item 1</placeholder-element>
          </pds-grid-item>
          <pds-grid-item>
            <placeholder-element>Item 2</placeholder-element>
          </pds-grid-item>
          <pds-grid-item>
            <placeholder-element>Item 3</placeholder-element>
          </pds-grid-item>
          <pds-grid-item>
            <placeholder-element>Item 4</placeholder-element>
          </pds-grid-item>
          <pds-grid-item>
            <placeholder-element>Item 5</placeholder-element>
          </pds-grid-item>
          <pds-grid-item>
            <placeholder-element>Item 6</placeholder-element>
          </pds-grid-item>
        </pds-grid>
      </pds-primary-navigation-main-menu-item>
      <pds-primary-navigation-main-menu-item
        href="#"
        text="Link"
      ></pds-primary-navigation-main-menu-item>
    </pds-primary-navigation-main-menu>`,
};
export default meta;

// span element is needed to wrap plain text slots https://github.com/storybookjs/storybook/issues/15963

export const Default: StoryObj = {
  name: 'Default',
  args: {
    variant: 'default',
    ariaLabel: 'main menu',
  },
  parameters: {
    a11y: {
      config: {
        rules: [
          {
            // Color contrast is sufficient - this is a false positive.
            //
            // @link https://dequeuniversity.com/rules/axe/4.6/color-contrast
            id: 'color-contrast',
            enabled: false,
          },
        ],
      },
    },
  },
};

export const LinkWithIcon: StoryObj = {
  name: 'Link with icon',
  args: {
    variant: 'default',
    ariaLabel: 'main menu',
  },
  parameters: {
    a11y: {
      config: {
        rules: [
          {
            // Color contrast is sufficient - this is a false positive.
            //
            // @link https://dequeuniversity.com/rules/axe/4.6/color-contrast
            id: 'color-contrast',
            enabled: false,
          },
        ],
      },
    },
  },
  render: (args) =>
    html`<pds-primary-navigation-main-menu
      variant="${args.variant}"
      ariaLabel=${args.ariaLabel}
    >
      <pds-primary-navigation-main-menu-item dropdown text="Dropdown">
        <pds-list size="sm" spacing="sm">
          <pds-list-item
            ><pds-primary-navigation-dropdown-link href="#"
              >This is a link</pds-primary-navigation-dropdown-link
            ></pds-list-item
          >
          <pds-list-item
            ><pds-primary-navigation-dropdown-link href="#"
              >This is a link</pds-primary-navigation-dropdown-link
            ></pds-list-item
          >
          <pds-list-item
            ><pds-primary-navigation-dropdown-link href="#"
              >This is a link</pds-primary-navigation-dropdown-link
            ></pds-list-item
          >
          <pds-list-item
            ><pds-primary-navigation-dropdown-link href="#"
              >This is a link</pds-primary-navigation-dropdown-link
            ></pds-list-item
          >
        </pds-list>
      </pds-primary-navigation-main-menu-item>
      <pds-primary-navigation-main-menu-item megamenu text="Megamenu">
        <pds-grid variant="1-3up" break="slower">
          <pds-grid-item>
            <placeholder-element>Item 1</placeholder-element>
          </pds-grid-item>
          <pds-grid-item>
            <placeholder-element>Item 2</placeholder-element>
          </pds-grid-item>
          <pds-grid-item>
            <placeholder-element>Item 3</placeholder-element>
          </pds-grid-item>
          <pds-grid-item>
            <placeholder-element>Item 4</placeholder-element>
          </pds-grid-item>
          <pds-grid-item>
            <placeholder-element>Item 5</placeholder-element>
          </pds-grid-item>
          <pds-grid-item>
            <placeholder-element>Item 6</placeholder-element>
          </pds-grid-item>
        </pds-grid>
      </pds-primary-navigation-main-menu-item>
      <pds-primary-navigation-main-menu-item
        arrow
        href="#"
        text="Link"
      ></pds-primary-navigation-main-menu-item>
    </pds-primary-navigation-main-menu>`,
};
