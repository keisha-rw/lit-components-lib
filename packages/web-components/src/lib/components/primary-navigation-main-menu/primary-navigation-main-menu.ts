import { html } from 'lit';
import { property } from 'lit/decorators.js';
import { ifDefined } from 'lit/directives/if-defined.js';
import { pdsCustomElement as customElement } from '../../decorators/pds-custom-element';
import { PdsElement } from '../PdsElement';
import styles from './primary-navigation-main-menu.scss?inline';

/**
 * @summary This component is a navigational element (nav) that holds an Un unordered list element (ul)
 *
 * @slot default Contains one or more primary nav menu items within the pds-primary-navigation-main-menu, restricted to pds-primary-navigation-main-menu-item elements
 */
@customElement('pds-primary-navigation-main-menu', {
  category: 'component',
  type: 'component',
  state: 'stable',
  styles,
})
export class PdsPrimaryNavigationMainMenu extends PdsElement {
  /**
   * Style variant
   * - default treatment uses a default background
   * - **inverted** variant changes to the inverted background
   */
  @property()
  variant: 'default' | 'inverted' = 'default';

  /**
   * Screen reader label for button
   */
  @property({ type: String })
  ariaLabel: string;

  /**
   * @internal
   */
  get classNames() {
    return {
      [this.variant]: !!this.variant,
    };
  }

  render() {
    return html`<div
      class=${this.getClass()}
      aria-label=${ifDefined(this.ariaLabel)}
    >
      <ul class="pds-c-primary-navigation-main-menu__list" role="list">
        <slot
          allowed-elements="pds-primary-navigation-main-menu-item"
          @slotchange=${this.handleSlotValidation}
        ></slot>
      </ul>
    </div>`;
  }
}
