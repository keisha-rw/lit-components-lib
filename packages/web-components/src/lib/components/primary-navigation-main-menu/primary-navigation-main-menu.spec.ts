import * as path from 'path';
import initStoryshots from '@storybook/addon-storyshots';
import { assert, fixture } from '@open-wc/testing';
import { PdsPrimaryNavigationMainMenu } from './primary-navigation-main-menu';
import createSnapshots from '../../../../test/utils/snapshots';

initStoryshots({
  storyKindRegex: /Primary navigation main menu/,
  framework: 'web-components',
  configPath: path.join(__dirname, '../../../../.storybook'),
  asyncJest: true,
  test: createSnapshots,
});

describe('PdsPrimaryNavigationMainMenu unit tests', () => {
  let element: Element;

  beforeEach(async () => {
    element = await fixture('<pds-primary-navigation-main-menu/>');
  });

  it('is an instance of PdsPrimaryNavigationMainMenu', () => {
    expect(element).toBeInstanceOf(PdsPrimaryNavigationMainMenu);
  });

  it('is accessible', async () => {
    await assert.isAccessible(element);
  });

  it('populates the slot correctly with type pds-primary-navigation-main-menu-item', async () => {
    const consoleError = jest
      .spyOn(console, 'error')
      .mockImplementation(() => {});
    element = await fixture(
      '<pds-primary-navigation-main-menu variant="default"><pds-primary-navigation-main-menu-item/></pds-primary-navigation-main-menu>',
    );
    expect(consoleError).not.toHaveBeenCalled();
    consoleError.mockRestore();
  });

  it('does not popluate the slot correctly with any element not type of pds-primary-navigation-main-menu-item', async () => {
    const consoleError = jest
      .spyOn(console, 'error')
      .mockImplementation(() => {});
    element = await fixture(
      '<pds-primary-navigation-main-menu variant="default"><div> <pds-list size="sm" spacing="sm"><pds-list-item><pds-primary-navigation-dropdown-link href="#">This is a link</pds-primary-navigation-dropdown-link></pds-list-item></pds-list>/div></pds-primary-navigation-main-menu>',
    );
    expect(element).toMatchSnapshot();
    expect(consoleError).toHaveBeenCalledTimes(1);
    consoleError.mockRestore();
  });
});
