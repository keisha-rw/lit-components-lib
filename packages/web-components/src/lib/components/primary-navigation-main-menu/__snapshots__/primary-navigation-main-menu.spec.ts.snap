// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PdsPrimaryNavigationMainMenu unit tests does not popluate the slot correctly with any element not type of pds-primary-navigation-main-menu-item 1`] = `
<pds-primary-navigation-main-menu
  variant="default"
/>
`;
